src/Badges.sol

```diff
@@ -112,21 +129,45 @@ contract Badges is
     return address(specDataHolder);
   }
 
-  function createSpecAsRaftHolder(string memory _specUri, uint256 _raftTokenId) external virtual {
-    require(specDataHolder.getRaftOwner(_raftTokenId) == msg.sender, "createSpecAsRaftHolder: unauthorized");
-    require(!specDataHolder.isSpecRegistered(_specUri), "createSpecAsRaftHolder: spec already registered");
-    specDataHolder.setSpecToRaft(_specUri, _raftTokenId);
-    emit SpecCreated(msg.sender, _specUri, _raftTokenId, specDataHolder.getRaftAddress());
-  }
-
-  function createSpecAsHolderOfAnyBadge(
+  function createSpec(
```

Comment:
Why not have external functions specific to each permission? that way it forces the client to be aware of what permissioned function they're calling. Wdyt?